### ISSUE TITLE
papers.labml.ai

### ISSUE THREAD

**Issue by hnipun** [karpathy/arxiv-sanity-lite#2]
We built [papers.labml.ai](https://papers.labml.ai) in May ([introductory tweet](https://twitter.com/labmlai/status/1395978352175902726)) to discover research papers based on popularity in Twitter. We were using arxiv-sanity to discover papers and I started this as a side project inspired by it (partly because it was down from time to time).

We worked on it on and off since May and have added a bunch of features, such as:
- Popular papers based on Tweets
- Link source codes, [annotated implementations](https://nn.labml.ai), videos, reddit and hacker news discussions, and other resources related to the paper
- Conferences ([iclr 2022](https://papers.labml.ai/papers/iclr_2022), [neurips 2021](https://papers.labml.ai/papers/neurips_2021))
- Short two-line summaries of the papers to quickly browse through lists of papers
- Similar papers based on language model embeddings

And we are working on something very similar to tags on sanity-lite (which we call lists).

We love to hear your feedback and suggestions. Thanks for releasing your work. 

<img width="700" alt="Screenshot 2021-11-14 at 10 24 45" src="https://user-images.githubusercontent.com/9856846/141668110-f6f083ca-f91c-4d3c-86ed-05bf77fc2721.png">
<img width="700" alt="Screenshot 2021-11-14 at 10 25 36" src="https://user-images.githubusercontent.com/9856846/141668114-ac423acd-31a5-456e-80b0-ec08fb63cc62.png">
<img width="700" alt="Screenshot 2021-11-14 at 10 27 01" src="https://user-images.githubusercontent.com/9856846/141668123-31bbab81-6488-44f5-b88e-06eb4513a716.png">

**Comment by karpathy** [comment]
Hi Nipun, it's fun to hear from you! I actually have https://papers.labml.ai/papers/weekly pinned in my toolbar and visit it regularly, great work on the site and I look forward to seeing where you take it!

**Comment by hnipun** [comment]
Happy to hear you are finding it useful. Mostly, we make improvements based on our personal needs. Let us know if you have any suggestions for improvements. Thanks.

**Comment by GeorvityLabs** [comment]
@karpathy @hnipun could you guys add a feature , where it filters out papers from arxiv that have "github" repo links.

That way only the papers with github code gets displayed on the screen (for folks who are looking for papers having implementation ready)

**Comment by karpathy** [comment]
@GeorvityLabs  This would potentially require downloading the full text of the paper, dramatically increasing the complexity. Currently we can afford to only scrape the abstracts and this is very helpful. So I don't believe this is easy sadly.

**Comment by GeorvityLabs** [comment]
@karpathy that makes sense.

In case of some papers . the authors include github repo links in their abstract, so scrapping just the abstract alone would work in those cases.

But in most other papers, the github repo links are usually under : Code & Data, Comments or the Abstract section (on arxiv.org).
So, if we manage to scrape these three sections separately it would be possible to implement the feature.

**Comment by GeorvityLabs** [comment]
I tried using the davinci-codex engine , to generate a python script , so far this is what it has come up with :
```

import webbrowser

#get input string from user
input_string = input("Enter a string: ")

#search arvix.org for that input string
search_string = "https://arxiv.org/search/?query=" + input_string + "&searchtype=all&source=header"

#filter search such that only papers with code is displayed
search_string = search_string + "&filter=has-official-code:y"

#print the search string
print(search_string)

#open the search string in a new tab
webbrowser.open_new_tab(search_string)

```

I was wondering if there is any &filter options that helps us check the Code  & Data , Abstract and Comments separately.

**Comment by subramanya1997** [comment]
I think github repo can be obtained from papeswithcode.com. I just added a github_links field to papers table

**Comment by GeorvityLabs** [comment]
@subramanya1997 not all papers have code up on paperswithcode.com , usually people link their github repos more often than via paperswithcode.com.

**Comment by subramanya1997** [comment]
@GeorvityLabs True. Some of them won't be mentioned in the paper but would be released later. I though it is better to have some than have none.

**Comment by GeorvityLabs** [comment]
@subramanya1997 true. But, usually after been reading both paperwithcode and arvix for years, one thing i noticed is , usually most github links don't make it to paperswithcode. but like you said , something is better.
